notebooks/config.ipynb

```diff
@@ -964,12 +971,860 @@
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "help(config.compare)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "908b17fe-ae2d-4964-a659-0481d063c037",
+   "metadata": {},
+   "source": [
+    "Consider the following config files, which have similar values, with the exception of `sent`'s value.\n",
+    "<!--cell 56-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 29,
+   "id": "0218e5de-2c25-4d7a-a7b6-0f05ad81afb2",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "&memo\n",
+      "  sender_id = \"{{ id }}\"\n",
+      "  message = \"{{ greeting }}, {{ recipient }}!\"\n",
+      "  sent = .FALSE.\n",
+      "/\n",
+      "----------------------------------------------\n",
+      "&memo\n",
+      "  sender_id = \"{{ id }}\"\n",
+      "  message = \"{{ greeting }}, {{ recipient }}!\"\n",
+      "  sent = .TRUE.\n",
+      "/\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "cat fixtures/config/base-config.nml\n",
+    "echo ----------------------------------------------\n",
+    "cat fixtures/config/alt-config.nml "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c4361a7c-9c16-4bc0-b3e6-985d85f1450c",
+   "metadata": {},
+   "source": [
+    "`compare()` returns `True` if the configs contain identical key-value pairs, and `False` otherwise. If a logger has been initialized, information is logged on which files are being compared and the values that differ, if any. Files are passed to `config_1_path` and `config_2_path` as a string filename or <a href=\"https://docs.python.org/3/library/pathlib.html#pathlib.Path\">Path</a> object. Corresponding optional formats may be passed using `config_1_format` and `config_2_format` and are only needed if the format suffix is not recognized.\n",
+    "<!--cell 58-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "id": "988b72bc-7983-4d24-9d37-c9ba32a31ac4",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "[2024-10-16T08:00:19]     INFO - fixtures/config/base-config.nml\n",
+      "[2024-10-16T08:00:19]     INFO + fixtures/config/alt-config.nml\n",
+      "[2024-10-16T08:00:19]     INFO ---------------------------------------------------------------------\n",
+      "[2024-10-16T08:00:19]     INFO memo:            sent:  - False + True\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "False"
+      ]
+     },
+     "execution_count": 30,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "config.compare(\n",
+    "    config_1_path=Path('fixtures/config/base-config.nml'),\n",
+    "    config_2_path='fixtures/config/alt-config.nml',\n",
+    "    config_1_format='nml',\n",
+    "    config_2_format='nml'\n",
+    ")"
+   ]
+  },
```

Comment:
What do you think about adding an example of when two files are trying to be compared of differing types so users can see that error?

From the docs: "Comparisons are supported only for configs of the same format, e.g. YAML vs YAML, Fortran namelist vs Fortran namelist, etc. uw will flag invalid comparisons"

i.e. I used existing fixture files to mock up an example
```
config.compare(
    config_1_path=Path('fixtures/config/get-config.yaml'),
    config_2_path=Path('fixtures/config/base-config.nml')
)

[2024-10-16T14:54:58]    ERROR Formats do not match: yaml vs nml
```

Reply:
Yeah, I agree this is a great example to include. I added it at the end of the `config.compare()` section.